src/Microsoft.AspNet.Razor/Parser/HtmlMarkupParser.Block.cs

```diff
@@ -471,11 +471,18 @@ private void BeforeAttribute()
                 return;
             }
 
+            // Minimized attribute
             if (!At(HtmlSymbolType.Equals))
             {
-                // Saw a space or newline after the name, so just skip this attribute and continue around the loop
-                Accept(whitespace);
-                Accept(name);
+                Output(SpanKind.Markup);
+
+                using (Context.StartBlock(BlockType.Markup))
+                {
+                    Accept(whitespace);
+                    Accept(name);
+                    Output(SpanKind.Markup);
+                }
+
                 return;
             }
 
```

Comment:
Add a comment right below here reminding user that we're in the mainline scenario at this point i.e. recovery and minimized attribute cases were handled above.